src/main/java/racing/domain/Car.java

```diff
@@ -0,0 +1,57 @@
+package racing.domain;
+
+public class Car {
+    private static final int VALIDATED_NAME_LENGTH = 5;
+    private static final int MOVED_DISTANCE = 1;
+    private static final int NOT_MOVED_DISTANCE = 0;
+
+    private String name;
+    private MovedLog movedLog;
+    private MoveConditionStrategy moveConditionStrategy;
+
+    public Car(String name, MoveConditionStrategy moveConditionStrategy) {
+        validateLength(name);
+        this.name = name;
+        this.moveConditionStrategy = moveConditionStrategy;
+        this.movedLog = new MovedLog();
+    }
+
+    public Car(String name, MoveConditionStrategy moveConditionStrategy, int initDistance) {
+        validateLength(name);
+        this.name = name;
+        this.moveConditionStrategy = moveConditionStrategy;
+        this.movedLog = new MovedLog(initDistance);
+    }
+
+    private void validateLength(String name) {
+        if (name.length() > VALIDATED_NAME_LENGTH) {
+            throw new IllegalArgumentException("이름은 5자 이내로 입력하세요.");
+        }
+    }
+
+    public void move() {
+        boolean canMove = moveConditionStrategy.isMovable();
+        if (canMove) {
+            movedLog.addMovedLog(MOVED_DISTANCE);
+        }
+        if (!canMove) {
+            movedLog.addMovedLog(NOT_MOVED_DISTANCE);
+        }
+    }
```

Comment:
이 부분은 아래와 같이 변경해볼 수 있을 것 같아요. :)
```suggestion
        if (canMove) {
            movedLog.addMovedLog(MOVED_DISTANCE);
            return;
        }
        movedLog.addMovedLog(NOT_MOVED_DISTANCE);
```

추가적으로, 이동여부와 관계 없이 자동차 경주의 각 회차별로 자동차의 위치만 기록해도 괜찮을 것 같다는 생각이 드네요! 고민해보면 좋을 것 같네요! :)

Reply:
이 부분은 두가지의 장점을 판단하여 구현 하면 좋을 것 같아요 😀
예시가 정확할지 모르겠지만 대강 생각해보면

1. 이동여부를 판별해서 다르게 기록하는 경우 -> 전체 라운드에서 이동여부에 따른 거리가 달라 질 수있는경우(1칸대신 2칸,3칸)? 이동여부보다 거리가 중요하여 거리에 관련된 로직이 많을때

2. 이동여부 자체를 기록하는 경우  -> 이동 여부가 더 중요한 경우? 각 라운드별로 자동차가 이동했는지 안했는지 판단하여 bool 타입을 사용할 로직이 많을때(최종라운드에서 이동하면 추가거리보너스(3칸)를 준다?)
정도로 나눌 수 있지 않을까 싶어요